onnxruntime/contrib_ops/cuda/bert/group_query_attention_impl.cu

```diff
@@ -443,42 +543,78 @@
 
   bool is_causal = parameters.is_unidirectional;
 
-  if (data.past_key != nullptr && data.past_key == data.present_key) {
+  // Note: seqlens_k is past sequence length for flash
```

Comment:
we have 2 seqlens_k here: one is data.seqlens_k, the other is data_selens_k_buff. I think we can make it more clean. 
data.seqlens_k to store the total_sequence - 1 and data.seqlens_k_buff to store the total_sequence (also change its name to data.seqlens_k_total). It will make the handling below easier.